uosc.lua

```diff
@@ -2985,9 +2985,27 @@ function create_select_tracklist_type_menu_opener(menu_title, track_type, track_
 			if track.type == track_type then
 				if track.selected then active_item = track.id end
 
+				local hint_vals = {
+					track.lang and track.lang:upper() or nil,
+					track['demux-h'] and (track['demux-w'] and track['demux-w'] .. 'x' .. track['demux-h']
+					                      or track['demux-h'] .. 'p'),
+					track['demux-fps'] and string.format('%.5gfps', track['demux-fps']) or nil,
+					track.codec,
+					track['audio-channels'] and track['audio-channels'] .. ' channels' or nil,
+					track['demux-samplerate'] and string.format('%.3gkHz', track['demux-samplerate']/1000) or nil,
```

Comment:
I was just wondering if a utility function streamline this kind of "process if not nil" behavior would be helpful.

My Lua is a bit rusty, but something like this:

```lua
-- function first so that it can handle varargs
function ifnotnil(x, ...)
    local args = {...}
    for i = 1, #args do
        if not hint_vals[i] then
            return nil
        end
    end
    if (type(x) == 'function') then
        return x(...)
    else
        return x
    end
end

hint_vals = {
    ifnotnil(string.upper, track.lang),
    ifnotnil(function(w, h) return string.format('%sx%s', w, h); end, track['demux-w'], track['demux-h'])
        or ifnotnil(function(h) return h .. 'p' end, track['demux-h']),
    ifnotnil(function(fps) return string.format('%.5gfps', fps); end, track['demux-fps']),
    track.codec,
    ifnotnil(function(c) return c .. ' channels'; end, track['audio-channels']),
    ifnotnil(function(rate) return string.format('%.3gkHz', rate/1000); end, track['demux-samplerate']),
    ifnotnil('forced', track.forced),
    ifnotnil('Default', track.default),
}
```

Reply:
Yeah it' unfortunate that Lua doesn't have a more concise lambda syntax (and Moonscript has a whole other bag of issues).